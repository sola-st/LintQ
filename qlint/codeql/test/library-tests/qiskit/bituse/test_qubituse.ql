import python
import qiskit.BitUse
import qiskit.Circuit

from BitUse bu
where not bu.getLocation().getFile().getAbsolutePath().matches("%site-packages/qiskit/%")
select bu, bu.getAGate(),
  "bit use : circuit: " + bu.getACircuitName() + " - reg: " + bu.getARegisterName() + " - index: " +
    bu.getAnIndex() + " - gate: " + bu.getAGateName()
