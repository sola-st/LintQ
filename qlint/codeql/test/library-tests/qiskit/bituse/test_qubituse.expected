| concept_complex_bit_use.py:6:5:6:22 | ControlFlowNode for QuantumRegister() | concept_complex_bit_use.py:12:1:12:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 0 - gate: h |
| concept_complex_bit_use.py:6:5:6:22 | ControlFlowNode for QuantumRegister() | concept_complex_bit_use.py:12:1:12:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 1 - gate: h |
| concept_complex_bit_use.py:6:5:6:22 | ControlFlowNode for QuantumRegister() | concept_complex_bit_use.py:12:1:12:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 2 - gate: h |
| concept_complex_bit_use.py:6:5:6:22 | ControlFlowNode for QuantumRegister() | concept_complex_bit_use.py:12:1:12:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 3 - gate: h |
| concept_complex_bit_use.py:9:6:9:9 | ControlFlowNode for Subscript | concept_complex_bit_use.py:9:1:9:10 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 0 - gate: h |
| concept_complex_bit_use.py:10:7:10:10 | ControlFlowNode for Subscript | concept_complex_bit_use.py:10:1:10:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 0 - gate: cx |
| concept_complex_bit_use.py:10:13:10:16 | ControlFlowNode for Subscript | concept_complex_bit_use.py:10:1:10:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 1 - gate: cx |
| concept_complex_bit_use.py:11:12:11:15 | ControlFlowNode for Subscript | concept_complex_bit_use.py:11:1:11:16 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: q - index: 3 - gate: rx |
| concept_complex_bit_use.py:16:12:16:12 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:16:1:16:13 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 0 - gate: s |
| concept_complex_bit_use.py:17:13:17:13 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:17:1:17:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 0 - gate: cx |
| concept_complex_bit_use.py:17:16:17:16 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:17:1:17:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 1 - gate: cx |
| concept_complex_bit_use.py:18:13:18:13 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:18:1:18:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 1 - gate: cx |
| concept_complex_bit_use.py:18:16:18:16 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:18:1:18:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 2 - gate: cx |
| concept_complex_bit_use.py:19:13:19:13 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:19:1:19:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 2 - gate: cx |
| concept_complex_bit_use.py:19:16:19:16 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:19:1:19:17 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 3 - gate: cx |
| concept_complex_bit_use.py:20:12:20:12 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:20:1:20:13 | ControlFlowNode for Attribute() | bit use : circuit: qc_front - reg: anonymous register - index: 3 - gate: t |
| concept_complex_bit_use.py:37:20:37:20 | ControlFlowNode for IntegerLiteral | concept_complex_bit_use.py:37:11:37:17 | ControlFlowNode for HGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: HGate |
| concept_complex_bit_use.py:38:6:38:13 | ControlFlowNode for Subscript | concept_complex_bit_use.py:38:1:38:14 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qregA - index: 1 - gate: y |
| concept_gate_specification.py:9:6:9:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:9:1:9:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: x |
| concept_gate_specification.py:10:6:10:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:10:1:10:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: y |
| concept_gate_specification.py:11:6:11:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:11:1:11:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: z |
| concept_gate_specification.py:12:6:12:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:12:1:12:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: h |
| concept_gate_specification.py:13:6:13:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:13:1:13:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: s |
| concept_gate_specification.py:14:8:14:8 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:14:1:14:9 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: sdg |
| concept_gate_specification.py:15:6:15:6 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:15:1:15:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 6 - gate: t |
| concept_gate_specification.py:16:8:16:8 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:16:1:16:9 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 7 - gate: tdg |
| concept_gate_specification.py:17:24:17:24 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:17:1:17:25 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: rx |
| concept_gate_specification.py:18:24:18:24 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:18:1:18:25 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: ry |
| concept_gate_specification.py:19:22:19:22 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:19:1:19:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: rz |
| concept_gate_specification.py:20:24:20:24 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:20:1:20:25 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: u1 |
| concept_gate_specification.py:21:23:21:23 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:21:1:21:24 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: p |
| concept_gate_specification.py:22:31:22:31 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:22:1:22:32 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: u2 |
| concept_gate_specification.py:24:42:24:42 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:24:1:24:43 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: u3 |
| concept_gate_specification.py:25:7:25:13 | ControlFlowNode for Subscript | concept_gate_specification.py:25:1:25:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 0 - gate: cx |
| concept_gate_specification.py:25:16:25:22 | ControlFlowNode for Subscript | concept_gate_specification.py:25:1:25:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 1 - gate: cx |
| concept_gate_specification.py:26:7:26:13 | ControlFlowNode for Subscript | concept_gate_specification.py:26:1:26:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 1 - gate: cy |
| concept_gate_specification.py:26:16:26:22 | ControlFlowNode for Subscript | concept_gate_specification.py:26:1:26:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 2 - gate: cy |
| concept_gate_specification.py:27:7:27:13 | ControlFlowNode for Subscript | concept_gate_specification.py:27:1:27:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 2 - gate: cz |
| concept_gate_specification.py:27:16:27:22 | ControlFlowNode for Subscript | concept_gate_specification.py:27:1:27:23 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 3 - gate: cz |
| concept_gate_specification.py:28:21:28:27 | ControlFlowNode for Subscript | concept_gate_specification.py:28:1:28:50 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 3 - gate: ch |
| concept_gate_specification.py:28:43:28:49 | ControlFlowNode for Subscript | concept_gate_specification.py:28:1:28:50 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 4 - gate: ch |
| concept_gate_specification.py:29:33:29:39 | ControlFlowNode for Subscript | concept_gate_specification.py:29:1:29:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 4 - gate: crz |
| concept_gate_specification.py:29:55:29:61 | ControlFlowNode for Subscript | concept_gate_specification.py:29:1:29:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 5 - gate: crz |
| concept_gate_specification.py:30:33:30:39 | ControlFlowNode for Subscript | concept_gate_specification.py:30:1:30:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 4 - gate: crx |
| concept_gate_specification.py:30:55:30:61 | ControlFlowNode for Subscript | concept_gate_specification.py:30:1:30:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 5 - gate: crx |
| concept_gate_specification.py:31:13:31:13 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:31:1:31:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: crx |
| concept_gate_specification.py:31:16:31:16 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:31:1:31:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: crx |
| concept_gate_specification.py:32:33:32:39 | ControlFlowNode for Subscript | concept_gate_specification.py:32:1:32:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 4 - gate: cry |
| concept_gate_specification.py:32:55:32:61 | ControlFlowNode for Subscript | concept_gate_specification.py:32:1:32:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 5 - gate: cry |
| concept_gate_specification.py:33:33:33:39 | ControlFlowNode for Subscript | concept_gate_specification.py:33:1:33:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 5 - gate: cu1 |
| concept_gate_specification.py:33:55:33:61 | ControlFlowNode for Subscript | concept_gate_specification.py:33:1:33:62 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 6 - gate: cu1 |
| concept_gate_specification.py:34:32:34:38 | ControlFlowNode for Subscript | concept_gate_specification.py:34:1:34:61 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 6 - gate: cp |
| concept_gate_specification.py:34:54:34:60 | ControlFlowNode for Subscript | concept_gate_specification.py:34:1:34:61 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 7 - gate: cp |
| concept_gate_specification.py:35:51:35:57 | ControlFlowNode for Subscript | concept_gate_specification.py:35:1:35:80 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 6 - gate: cu3 |
| concept_gate_specification.py:35:73:35:79 | ControlFlowNode for Subscript | concept_gate_specification.py:35:1:35:80 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 7 - gate: cu3 |
| concept_gate_specification.py:36:61:36:67 | ControlFlowNode for Subscript | concept_gate_specification.py:36:1:36:90 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 6 - gate: cu |
| concept_gate_specification.py:36:83:36:89 | ControlFlowNode for Subscript | concept_gate_specification.py:36:1:36:90 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 7 - gate: cu |
| concept_gate_specification.py:37:9:37:9 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:37:1:37:13 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: swap |
| concept_gate_specification.py:37:12:37:12 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:37:1:37:13 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: swap |
| concept_gate_specification.py:38:8:38:8 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:38:1:38:15 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: ccx |
| concept_gate_specification.py:38:11:38:11 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:38:1:38:15 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: ccx |
| concept_gate_specification.py:38:14:38:14 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:38:1:38:15 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: ccx |
| concept_gate_specification.py:39:10:39:10 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:39:1:39:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: cswap |
| concept_gate_specification.py:39:13:39:13 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:39:1:39:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: cswap |
| concept_gate_specification.py:39:16:39:16 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:39:1:39:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: cswap |
| concept_gate_specification.py:40:26:40:26 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:40:1:40:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: rxx |
| concept_gate_specification.py:40:36:40:36 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:40:1:40:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: rxx |
| concept_gate_specification.py:41:26:41:26 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:41:1:41:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: ryy |
| concept_gate_specification.py:41:36:41:36 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:41:1:41:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: ryy |
| concept_gate_specification.py:42:26:42:26 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:42:1:42:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: rzz |
| concept_gate_specification.py:42:36:42:36 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:42:1:42:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: rzz |
| concept_gate_specification.py:43:26:43:26 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:43:1:43:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: rzx |
| concept_gate_specification.py:43:36:43:36 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:43:1:43:37 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: rzx |
| concept_gate_specification.py:44:13:44:13 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:44:1:44:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: rzz |
| concept_gate_specification.py:44:16:44:16 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:44:1:44:17 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: rzz |
| concept_gate_specification.py:45:12:45:18 | ControlFlowNode for Subscript | concept_gate_specification.py:45:1:45:28 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qreg - index: 0 - gate: measure |
| concept_gate_specification.py:49:27:49:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:49:11:49:17 | ControlFlowNode for XGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: XGate |
| concept_gate_specification.py:50:27:50:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:50:11:50:17 | ControlFlowNode for YGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: YGate |
| concept_gate_specification.py:51:27:51:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:51:11:51:17 | ControlFlowNode for ZGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: ZGate |
| concept_gate_specification.py:52:27:52:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:52:11:52:17 | ControlFlowNode for HGate() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: HGate |
| concept_gate_specification.py:53:27:53:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:53:11:53:17 | ControlFlowNode for SGate() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: SGate |
| concept_gate_specification.py:54:29:54:29 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:54:11:54:19 | ControlFlowNode for SdgGate() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: SdgGate |
| concept_gate_specification.py:55:27:55:27 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:55:11:55:17 | ControlFlowNode for TGate() | bit use : circuit: qc - reg: anonymous register - index: 6 - gate: TGate |
| concept_gate_specification.py:56:29:56:29 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:56:11:56:19 | ControlFlowNode for TdgGate() | bit use : circuit: qc - reg: anonymous register - index: 7 - gate: TdgGate |
| concept_gate_specification.py:57:37:57:37 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:57:11:57:27 | ControlFlowNode for RXGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: RXGate |
| concept_gate_specification.py:58:31:58:31 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:58:11:58:27 | ControlFlowNode for RYGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: RYGate |
| concept_gate_specification.py:59:29:59:29 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:59:11:59:25 | ControlFlowNode for RZGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: RZGate |
| concept_gate_specification.py:60:31:60:31 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:60:11:60:27 | ControlFlowNode for U1Gate() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: U1Gate |
| concept_gate_specification.py:61:38:61:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:61:11:61:34 | ControlFlowNode for U2Gate() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: U2Gate |
| concept_gate_specification.py:62:49:62:49 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:62:11:62:45 | ControlFlowNode for U3Gate() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: U3Gate |
| concept_gate_specification.py:63:28:63:34 | ControlFlowNode for Subscript | concept_gate_specification.py:63:11:63:18 | ControlFlowNode for CXGate() | bit use : circuit: qc - reg: qreg - index: 0 - gate: CXGate |
| concept_gate_specification.py:63:37:63:43 | ControlFlowNode for Subscript | concept_gate_specification.py:63:11:63:18 | ControlFlowNode for CXGate() | bit use : circuit: qc - reg: qreg - index: 1 - gate: CXGate |
| concept_gate_specification.py:64:28:64:34 | ControlFlowNode for Subscript | concept_gate_specification.py:64:11:64:18 | ControlFlowNode for CYGate() | bit use : circuit: qc - reg: qreg - index: 1 - gate: CYGate |
| concept_gate_specification.py:64:37:64:43 | ControlFlowNode for Subscript | concept_gate_specification.py:64:11:64:18 | ControlFlowNode for CYGate() | bit use : circuit: qc - reg: qreg - index: 2 - gate: CYGate |
| concept_gate_specification.py:65:28:65:34 | ControlFlowNode for Subscript | concept_gate_specification.py:65:11:65:18 | ControlFlowNode for CZGate() | bit use : circuit: qc - reg: qreg - index: 2 - gate: CZGate |
| concept_gate_specification.py:65:37:65:43 | ControlFlowNode for Subscript | concept_gate_specification.py:65:11:65:18 | ControlFlowNode for CZGate() | bit use : circuit: qc - reg: qreg - index: 3 - gate: CZGate |
| concept_gate_specification.py:66:28:66:28 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:66:11:66:18 | ControlFlowNode for CHGate() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: CHGate |
| concept_gate_specification.py:66:31:66:31 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:66:11:66:18 | ControlFlowNode for CHGate() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: CHGate |
| concept_gate_specification.py:67:50:67:56 | ControlFlowNode for Subscript | concept_gate_specification.py:67:23:67:40 | ControlFlowNode for CRZGate() | bit use : circuit: qc - reg: qreg - index: 4 - gate: CRZGate |
| concept_gate_specification.py:67:59:67:65 | ControlFlowNode for Subscript | concept_gate_specification.py:67:23:67:40 | ControlFlowNode for CRZGate() | bit use : circuit: qc - reg: qreg - index: 5 - gate: CRZGate |
| concept_gate_specification.py:68:50:68:56 | ControlFlowNode for Subscript | concept_gate_specification.py:68:23:68:40 | ControlFlowNode for CU1Gate() | bit use : circuit: qc - reg: qreg - index: 5 - gate: CU1Gate |
| concept_gate_specification.py:68:59:68:65 | ControlFlowNode for Subscript | concept_gate_specification.py:68:23:68:40 | ControlFlowNode for CU1Gate() | bit use : circuit: qc - reg: qreg - index: 6 - gate: CU1Gate |
| concept_gate_specification.py:69:42:69:42 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:69:11:69:32 | ControlFlowNode for CU3Gate() | bit use : circuit: qc - reg: anonymous register - index: 6 - gate: CU3Gate |
| concept_gate_specification.py:69:45:69:45 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:69:11:69:32 | ControlFlowNode for CU3Gate() | bit use : circuit: qc - reg: anonymous register - index: 7 - gate: CU3Gate |
| concept_gate_specification.py:70:30:70:30 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:70:11:70:20 | ControlFlowNode for SwapGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: SwapGate |
| concept_gate_specification.py:70:33:70:33 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:70:11:70:20 | ControlFlowNode for SwapGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: SwapGate |
| concept_gate_specification.py:71:29:71:29 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:71:11:71:19 | ControlFlowNode for CCXGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: CCXGate |
| concept_gate_specification.py:71:32:71:32 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:71:11:71:19 | ControlFlowNode for CCXGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: CCXGate |
| concept_gate_specification.py:71:35:71:35 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:71:11:71:19 | ControlFlowNode for CCXGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: CCXGate |
| concept_gate_specification.py:72:31:72:31 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:72:11:72:21 | ControlFlowNode for CSwapGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: CSwapGate |
| concept_gate_specification.py:72:34:72:34 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:72:11:72:21 | ControlFlowNode for CSwapGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: CSwapGate |
| concept_gate_specification.py:72:37:72:37 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:72:11:72:21 | ControlFlowNode for CSwapGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: CSwapGate |
| concept_gate_specification.py:73:38:73:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:73:11:73:28 | ControlFlowNode for RXXGate() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: RXXGate |
| concept_gate_specification.py:73:41:73:41 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:73:11:73:28 | ControlFlowNode for RXXGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: RXXGate |
| concept_gate_specification.py:74:38:74:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:74:11:74:28 | ControlFlowNode for RYYGate() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: RYYGate |
| concept_gate_specification.py:74:41:74:41 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:74:11:74:28 | ControlFlowNode for RYYGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: RYYGate |
| concept_gate_specification.py:75:38:75:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:75:11:75:28 | ControlFlowNode for RZZGate() | bit use : circuit: qc - reg: anonymous register - index: 2 - gate: RZZGate |
| concept_gate_specification.py:75:41:75:41 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:75:11:75:28 | ControlFlowNode for RZZGate() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: RZZGate |
| concept_gate_specification.py:76:38:76:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:76:11:76:28 | ControlFlowNode for RZXGate() | bit use : circuit: qc - reg: anonymous register - index: 3 - gate: RZXGate |
| concept_gate_specification.py:76:41:76:41 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:76:11:76:28 | ControlFlowNode for RZXGate() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: RZXGate |
| concept_gate_specification.py:77:38:77:38 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:77:11:77:28 | ControlFlowNode for RZZGate() | bit use : circuit: qc - reg: anonymous register - index: 4 - gate: RZZGate |
| concept_gate_specification.py:77:41:77:41 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:77:11:77:28 | ControlFlowNode for RZZGate() | bit use : circuit: qc - reg: anonymous register - index: 5 - gate: RZZGate |
| concept_gate_specification.py:78:29:78:29 | ControlFlowNode for IntegerLiteral | concept_gate_specification.py:78:11:78:19 | ControlFlowNode for Measure() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: Measure |
| concept_measure.py:3:6:3:23 | ControlFlowNode for QuantumRegister() | concept_measure.py:21:1:21:18 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 0 - gate: measure |
| concept_measure.py:3:6:3:23 | ControlFlowNode for QuantumRegister() | concept_measure.py:21:1:21:18 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 1 - gate: measure |
| concept_measure.py:8:12:8:16 | ControlFlowNode for Subscript | concept_measure.py:8:1:8:24 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 0 - gate: measure |
| concept_measure.py:11:12:11:16 | ControlFlowNode for Subscript | concept_measure.py:11:1:11:20 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 0 - gate: measure |
| concept_measure.py:12:12:12:12 | ControlFlowNode for IntegerLiteral | concept_measure.py:12:1:12:20 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: measure |
| concept_measure.py:15:6:15:6 | ControlFlowNode for IntegerLiteral | concept_measure.py:15:1:15:7 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: h |
| concept_measure.py:16:12:16:12 | ControlFlowNode for IntegerLiteral | concept_measure.py:16:1:16:16 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 0 - gate: measure |
| concept_measure.py:19:6:19:10 | ControlFlowNode for Subscript | concept_measure.py:19:1:19:11 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 0 - gate: h |
| concept_measure.py:20:7:20:11 | ControlFlowNode for Subscript | concept_measure.py:20:1:20:15 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: qr - index: 0 - gate: cx |
| concept_measure.py:20:14:20:14 | ControlFlowNode for IntegerLiteral | concept_measure.py:20:1:20:15 | ControlFlowNode for Attribute() | bit use : circuit: qc - reg: anonymous register - index: 1 - gate: cx |
